src/Raven.Server/Documents/Handlers/Processors/Batches/AbstractBatchHandlerProcessorForBulkDocs.cs

```diff
@@ -238,57 +239,64 @@ protected override void OnFinalize()
             WaitForSpecificIndexes = ConvertToStringValues("waitForSpecificIndex");
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         protected override void OnValue(QueryStringEnumerable.EncodedNameValuePair pair)
         {
             var name = pair.EncodedName;
 
-            if (IsMatch(name, NoReplyQueryStringName))
+            switch (name.Length)
             {
-                NoReply = GetBoolValue(name, pair.EncodedValue);
-                return;
-            }
-
-            if (IsMatch(name, WaitForIndexesTimeoutQueryStringName))
-            {
-                WaitForIndexesTimeout = GetTimeSpan(name, pair.EncodedValue);
-                return;
-            }
-
-            if (IsMatch(name, WaitForIndexThrowQueryStringName))
-            {
-                WaitForIndexThrow = GetBoolValue(name, pair.EncodedValue);
-                return;
-            }
-
-            if (IsMatch(name, WaitForSpecificIndexQueryStringName))
-            {
-                AddForStringValues("waitForSpecificIndex", pair.DecodeValue());
-                return;
-            }
-
-            if (IsMatch(name, WaitForReplicasTimeoutQueryStringName))
-            {
-                WaitForReplicasTimeout = GetTimeSpan(name, pair.EncodedValue);
-                return;
-            }
-
-            if (IsMatch(name, NumberOfReplicasToWaitForQueryStringName))
-            {
-                var value = pair.DecodeValue();
-                if (IsMatch(value, MajorityValue))
+                case 7:
```

Comment:
Can we use `NoReplyQueryStringName.Lenght` instead of 7? Here in all other places?

Reply:
No, because it is not a constant. However, I will see if I can do something better here... Maybe we can leverage Roslyn and the JIT for it.

Reply:
It's not a big deal. It will be verified by tests anyway. So maybe it's not worth it.

Reply:
Being looking into it anyways, none of the ideas I had made the code really easier to follow anyways.